core/scripts/keystone/src/99_files.go

```diff
@@ -1,71 +1,54 @@
 package src
 
 import (
-	"bufio"
 	"encoding/json"
 	"fmt"
 	"io"
 	"os"
-
-	"github.com/smartcontractkit/chainlink/v2/core/utils"
 )
 
 const (
 	defaultArtefactsDir   = "artefacts"
-	defaultPublicKeys     = ".cache/PublicKeys.json"
-	defaultNodeList       = ".cache/NodeList.txt"
+	defaultNodeSetsPath   = ".cache/node_sets.json"
 	deployedContractsJSON = "deployed_contracts.json"
-	bootstrapSpecTemplate = "bootstrap.toml"
-	cribOverrideTemplate  = "crib-overrides.yaml"
-	oracleSpecTemplate    = "oracle.toml"
 )
 
-func writeLines(lines []string, path string) error {
-	file, err := os.Create(path)
+func mustReadJSON[T any](fileName string) (output T) {
+	jsonFile, err := os.Open(fileName)
 	if err != nil {
-		return err
-	}
-	wc := utils.NewDeferableWriteCloser(file)
-	defer wc.Close()
-
-	w := bufio.NewWriter(file)
-	for _, line := range lines {
-		fmt.Fprintln(w, line)
-	}
-	if err := w.Flush(); err != nil {
-		return err
+		panic(fmt.Sprintf("failed to open file at %s: %v", fileName, err))
 	}
-	return wc.Close()
-}
-
-func readLines(path string) ([]string, error) {
-	file, err := os.Open(path)
+	defer jsonFile.Close()
+	bytes, err := io.ReadAll(jsonFile)
 	if err != nil {
-		return nil, err
+		panic(fmt.Sprintf("failed to read file at %s: %v", fileName, err))
 	}
-	defer file.Close()
-
-	var lines []string
-	scanner := bufio.NewScanner(file)
-	for scanner.Scan() {
-		lines = append(lines, scanner.Text())
+	err = json.Unmarshal(bytes, &output)
+	if err != nil {
+		panic(fmt.Sprintf("failed to unmarshal data: %v", err))
 	}
-	return lines, scanner.Err()
+	return
 }
 
-func mustParseJSON[T any](fileName string) (output T) {
-	jsonFile, err := os.Open(fileName)
+func mustWriteJSON[T any](fileName string, data T) {
+	jsonFile, err := os.Create(fileName)
 	if err != nil {
-		panic(err)
+		panic(fmt.Sprintf("failed to create file at %s: %v", fileName, err))
 	}
 	defer jsonFile.Close()
-	bytes, err := io.ReadAll(jsonFile)
+	encoder := json.NewEncoder(jsonFile)
+	encoder.SetIndent("", " ")
+	err = encoder.Encode(data)
 	if err != nil {
-		panic(err)
+		panic(fmt.Sprintf("failed to encode data: %v", err))
 	}
-	err = json.Unmarshal(bytes, &output)
+}
+
+func ensureArtefactsDir(artefactsDir string) {
+	_, err := os.Stat(artefactsDir)
 	if err != nil {
```

Comment:
Should this be checking for a certain kind of error? `https://pkg.go.dev/os#IsNotExist`